### ISSUE TITLE
Angular CLI (8 & 9) generated app re-compiles when opening a different file in VS Code

### ISSUE THREAD

**Issue by vitale232** [microsoft/vscode#106954]
- VSCode Version: 1.50.0-insider (user setup)
- OS Version: Windows 10 Enterprise v1909 OS build 1836.1016

### Steps to Reproduce:

1. Create a new Angular 9 application using the Angular CLI. Follow the brief [Lazy Loading tutorial](https://angular.io/guide/lazy-loading-ngmodules#step-by-step) to generate an application with multiple "Feature Modules" that will be chunked.
2. Change the currently viewed file in VS code. The application recompiles without any changes to the source, and without any "save" operations in the project directory.

### Does this issue occur when all extensions are disabled?: Yes/No
When using the flag as instructed in this issues template (`code-insiders --disable-extensions`), no, the issue cannot be replicated. However, there are no third party extensions installed, and the issue exists without the --disable-extensions flag.

### More Info
Here is a gif with a basic Angular 9 application that shows the issue:

![recompile-no-extensions](https://user-images.githubusercontent.com/8504254/93511643-1fa5fc80-f8f1-11ea-841d-3c4188d0915f.gif)

This can be a major nuisance on a real application, as the recompile time is much slower and any data you input while testing is obviously lost. This issue has been reported in this repo previously, but the issues I was able to uncover were closed due to lack of replication or lack of response. Some other issues/StackOverflow posts state that system backups can present this issue. I've confirmed with my IT Department that my project is located in a directory that is _excluded_ from their automated backups.

In the small repro here, I must admit that I can only replicate this intermittently. The reason I chose the Lazy Loading example is the issue seems to be worse when switching between files in different feature modules in our actual apps, and I could not replicate it with the default `hello-angular` app.

This issue has been driving my teammates to use Web Storm, as it can really block a developer's productivity.

**Comment by tedsync** [comment]
Same problem, very annoying.

**Comment by vitale232** [comment]
I was just playing with this a little more using VS Code Insiders.

It does seem like this problem is worsened when I install some extensions. I like to use the "Angular Essentials" extension pack (`johnpapa.angular-essentials`). When I installed that, it's a lot easier to replicate this issue.

**Comment by mjbvz** [comment]
@vitale232 So you do not see this when extensions are disabled? If the `angular essentials` extension causes this, can you file the issue against that extension (it's an extension pack so it would be even better if you could track down the specific extension that causes this)

**Comment by vitale232** [comment]
@mjbvz - Sorry if that last comment is confusing. I think this issue is _worse_ with extensions, but it's definitely happening with extensions disabled.

In this gif, I'm using `ng serve` via the VS Code terminal in one of the project applications I'm currently working on. I've pulled dev tools into the frame as well. According to this [StackOverflow](https://stackoverflow.com/questions/48605479/when-running-ng-serve-watch-how-does-the-browser-know-when-to-refresh-when-t), the Angular CLI is using a web socket connection to communicate file changes. I'm opening the editor by running this command from the project root: `code-insiders --disable-extensions .`. On this app, the problem is replicated easily and rapidly:

![reload-websocket](https://user-images.githubusercontent.com/8504254/93665669-580b1f00-fa46-11ea-9584-0b310457b0f7.gif)

In this gif, I'm using the Lazy Loading example again. I was having trouble replicating the issue, so I added a third Feature Module called invoices, and the issue was then pretty easy to replicate. 

![here-again](https://user-images.githubusercontent.com/8504254/93665814-89d0b580-fa47-11ea-818d-c03f36c0c646.gif)

Here's the code for this repo: https://github.com/vitale232/ng-serve-vs-code-reload-replication

I'm more than willing to produce log files or whatever else may be helpful in tracking this down. Just let me know. Thanks!

**Comment by vitale232** [comment]
Hey @mjbvz,

I just wanted to check in and see if you've had an opportunity to explore this ticket at all? This issue is still present for me in the latest release of VS Code.

```
Version: 1.49.2 (user setup)
Commit: e5e9e69aed6e1984f7499b7af85b3d05f9a6883a
Date: 2020-09-24T16:29:41.983Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.18363
```

Thanks!

**Comment by tedsync** [comment]
Any news about this please? It is a really annoying issue. No one develops in Angular using VSC?

**Comment by vitale232** [comment]
@mjbvz,

I have a little more information to share.

I've been using VSC/Angular on my company issued laptop, which runs Windows 10 and is heavily managed by our IT department. While researching this problem, I've come across a few issues that concluded live virus scans can often create this issue. My company policy has "Windows Defender Real Time Protection" enabled on the vast majority of my system.

I've worked with IT to try a couple of things, and we've pretty much ruled out Windows Defender as the cause of this issue. We've tried adding Node.exe and Code.exe as process exclusions, and the problem remains. We also just tried completely disabling the real-time protection, and the problem still persists there, too.

@tedsync - Are you on Windows 10? If so, can you try disabling your live scans as well to see if that helps?

**Comment by tedsync** [comment]
@vitale232 I am working in the same way than you. I'm on W10 and Windows Defender but unfortunately can't change anything on it due to IT rules. Thanks a lot for the option, it really helps :-)

**Comment by mjbvz** [comment]
Can you please share a minimal GitHub repro and steps to reproduce this

**Comment by vitale232** [comment]
@mjbvz  Please reopen this. I'll centralize the info for you:

### Steps to Reproduce:

1. Create a new Angular 9 application using the Angular CLI. Follow the brief [Lazy Loading tutorial](https://angular.io/guide/lazy-loading-ngmodules#step-by-step) to generate an application with multiple "Feature Modules" that will be chunked.
2. Change the currently viewed file in VS code. The application recompiles without any changes to the source, and without any "save" operations in the project directory.

Here's the code for this repo: https://github.com/vitale232/ng-serve-vs-code-reload-replication

**Comment by mjbvz** [comment]
This doesn't repo for me in VS Code insiders without extensions installed. Please share specific steps to reproduce the problem so I can confirm I am testing it correctly

**Comment by captainkout** [comment]
In my experiment, I noticed that it only recompile/reloads the FIRST time you open a file. I was able to reproduce this with the repo here https://github.com/vitale232/ng-serve-vs-code-reload-replication. and with all extensions disabled.

**Comment by vitale232** [comment]
@mjbvz I'm able to reproduce this quite easily with that repo on VSC Insiders with extensions disabled via the CLI argument.

Are you using Windows 10? You can see that all of the security settings are turned on, and I can't change them in these screenshots:

![image](https://user-images.githubusercontent.com/8504254/98724707-22a5f200-2362-11eb-84f2-13a71a19c07f.png)
![image](https://user-images.githubusercontent.com/8504254/98724739-2e91b400-2362-11eb-96cb-d2d950afc67e.png)

I'm of the opinion that some of the security settings could be part of the issue. In particular, I wish I could turn off the Real-time Protection feature for testing purposes. Part of the reason I'm weary of these settings is other IDEs (e.g. Web Storm) go as far as to warn you about these settings on startup:

![image](https://user-images.githubusercontent.com/8504254/98725139-c1cae980-2362-11eb-95fb-f23a0969aa7c.png)

**Comment by mjbvz** [comment]
Yes I tested on windows 10 using the repo you shared. I don't have a Windows environment with all those security settings. Can you try isolating to see if any of those settings is the cause of this issue?

I'll move this over to TypeScript to see if they have any ideas as well
